grails-app/conf/spring/resources.groovy

```diff
@@ -96,30 +99,65 @@ beans = {
+            if (transmartSecurity.ldap.newUsernamePattern) {
+                newUsernamePattern = transmartSecurity.ldap.newUsernamePattern
+            }
+            // comma separated list of new user authorities
+            if (transmartSecurity.ldap.defaultAuthorities) {
+                defaultAuthorities = transmartSecurity.ldap.defaultAuthorities
+            }
+            // if noPassword specified user will be created without password
+            if (transmartSecurity.ldap.noPassword) {
+                noPassword = transmartSecurity.ldap.noPassword
+            }
+            // can be 'username' or 'federatedId' (default: 'username')
+            if (transmartSecurity.ldap.mappedUsernameProperty) {
+                mappedUsernameProperty = transmartSecurity.ldap.mappedUsernameProperty
+            }
+        }
+
+        if (grailsApplication.config.org.transmart.security.ldap.ad.domain) {
+            xmlns aop:"http://www.springframework.org/schema/aop"
+
+            adExtension(ActiveDirectoryLdapAuthenticationExtension)
+
+            aop {
+                config("proxy-target-class": true) {
+                    aspect(id: 'adExtensionService', ref: 'adExtension')
+                }
+            }
+
+            ldapAuthProvider(ActiveDirectoryLdapAuthenticationProvider,
+                    transmartSecurity.ldap.ad.domain,
+                    SpringSecurityUtils.securityConfig.ldap.context.server
+            ) {
+                userDetailsContextMapper = ref('ldapUserDetailsMapper')
+            }
+        }
+    }
+
     if (grailsApplication.config.org.transmart.security.spnegoEnabled) {
         // plugin is not functional at this point
 //        SpnegoAuthenticationProcessingFilter(SpnegoAuthenticationProcessingFilter) {
 //            authenticationManager = ref('authenticationManager')
 //            failureHandler = ref('failureHandler')
 //        }
-
-        ldapUserDetailsMapper(com.recomdata.security.LdapAuthUserDetailsMapper) {
-            dataSource = ref('dataSource')
-            springSecurityService = ref('springSecurityService')
-            databasePortabilityService = ref('databasePortabilityService')
-            bruteForceLoginLockService = ref('bruteForceLoginLockService')
-        }
-
     } else {
         // plugin is not functional at this point
 //        SpringSecurityKerberosGrailsPlugin.metaClass.getDoWithSpring = {->
 //            logger.info "Skipped Kerberos Grails plugin initialization"
 //            return {}
 //        }
-        SpringSecurityLdapGrailsPlugin.metaClass.getDoWithSpring = { ->
-            logger.info "Skipped LDAP Grails plugin initialization"
-            return {}
-        }
+//        SpringSecurityLdapGrailsPlugin.metaClass.getDoWithSpring = { ->
+//            logger.info "Skipped LDAP Grails plugin initialization"
+//            return {}
+//        }
```

Comment:
In general, don't leave commented out code without explanation. In particular, why are you not skipping the initialization of the ldap plugin if ldap is not enable anymore? It seems to me that this should go to an else block following the `if (SpringSecurityUtils.securityConfig.ldap.active) {}`.